### PR TITLE
Added build for Android Play store bundle.

### LINKED ISSUE #1685: Mobile support?
Has there been any effort to port the configurator to a mobile (Android and possibly iOS) app? It seems that either Cordova or Flutter would be the best approach. It'll basically be a webview with a thin wrapper around the chrome.* calls. As far as I can see the following Chrome Apps APIs need to be ported:
```javascript
chrome.runtime.lastError
chrome.storage.local.get/set/remove
chrome.app.window.create/current
chrome.fileSystem.*
chrome.usb.*
chrome.serial.*
chrome.sockets.tcp.*
```
For Cordova, the following plugins already exist, although most of them are not actively maintained anymore:
https://github.com/MobileChromeApps/cordova-plugin-chrome-apps-runtime
https://github.com/MobileChromeApps/cordova-plugin-chrome-apps-storage
https://github.com/MobileChromeApps/cordova-plugin-chrome-apps-fileSystem
https://github.com/MobileChromeApps/cordova-plugin-chrome-apps-usb
https://github.com/r1000ru/cordova-plugin-chrome-apps-serial
https://github.com/KoenLav/cordova-plugin-chrome-apps-sockets-tcp

For Flutter or another framework, it would require creating the Chrome Apps API wrapper from scratch.

### PR DESCRIPTION
**Pull request by mikeller**
Fixes #1685.

Based on #2574, will need a rebase after this has been merged.

Included:
- [x] added support for debug builds for android, producing an APK file for local installation;
- [x] changed the release build for android to produce an AAB file for Play store upload;
- [x] changed the release build for android to use a private keystore;
- [x] changed the release build for android to ask for the version name to be used, to work around the limitations on versioning applications in the Google Play store.

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: gulpfile.js
Comment (DusKing1) [inline gulpfile.js:724]: Is this comment still needed?

Comment (mikeller) [reply gulpfile.js:724]: I haven't heard anything about dashes ('-') now being supported in RPM version names. Have you?

Comment (DusKing1) [reply gulpfile.js:724]: I see, I misunderstood this comment that I think this RPM is "Revolutions Per minute". It's the Red-Hat Package Manager, right?

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: azure-pipelines.yml
Comment (Asizon) [inline azure-pipelines.yml:89]: Now we need to have both here, debug and non debug build like in Windows:
```
    - script: yarn gulp release --android
      displayName: 'Run yarn release for android'
      condition: and(succeeded(), eq('${{ parameters.releaseBuild }}', true))
    - script: yarn gulp debug-release --android
      displayName: 'Run yarn debug release for android'
      condition: and(succeeded(), eq('${{ parameters.releaseBuild }}', false))
```

**Review by Asizon — CHANGES_REQUESTED** [review]

File: azure-pipelines.yml
Comment (mikeller) [reply azure-pipelines.yml:89]: The problem is that part of the build of the android application bundle for the Google Play store is to sign the bundle with a private key. This cannot easily moved to azure without compromising the key, so I have opted to be doing this on a physical machine for now.
There are some more constraints with the Google Play store around versioning that make it hard to fully automate builds to be published in this way, so doing this manually and locally is probably best anyway.

Comment (McGiverGim) [reply azure-pipelines.yml:89]: I don' t understand exactly... I understand the part that we need a private key, but I don't understand why we can't have the two options (release and debug-release) like in windows. I understand that by default it will execute the debug-release anyway.
I suppose that what you say is that the release will not work because it simply does not have the private key so is better to not put the option here?
If that is the reason, then what we do is to use different private keys for the "official" release and the "official" nightly, what does not seem very clean. The user will need to remove one to install the other. There must be a way to use, in a safe way, the same private key for both versions. Another thing is a local build of the developer, in this case of course the private key will be different only for testing.

**Review by McGiverGim — COMMENTED** [review]
Does the `cordova/release.keystore` contain the real private key for signing the final apk? It does not seem very secure to publish the file here then.

**Comment by McGiverGim** [comment]
I have been looking how others Android apps do this... for example here, is with a Github Action, but I suppose Azure will have something very similar:
This is a relase.xml workflow: https://github.com/home-assistant/android/blob/master/.github/workflows/release.yml , inside it, it takes the keystore and password from the github secrets and calls this action to inflate it: https://github.com/home-assistant/android/blob/master/.github/actions/inflate-secrets/action.yml In this way you need to upload the keystore and password to the secrets, but at least is not published in the public github repo.

**Comment by mikeller** [comment]
@McGiverGim: 
> Does the cordova/release.keystore contain the real private key for signing the final apk? It does not seem very secure to publish the file here then.

No, releases use `bundle.keystore`, which isn't checked in:

https://github.com/betaflight/betaflight-configurator/pull/2577/files#diff-7d841060297ed1d22543caa294064687c58a2a38cf9f13b809d64b61db0a5db7R11

**Comment by McGiverGim** [comment]
> What value do you see in getting the application bundle that Google uses to build the application that are distributed through the Google Play store built in the cloud - there is no way for a user to verify that his application was not tampered with anyway.

Not for the final version. A lot of projects automate the release, but I understand that we do the release manually, and I'm not against it. But I think it can be valuable to push the nightly version. The user can decide if they want to be subscribed to the beta channel of the app, or only to the release channel. If subscribed to the beta, I think we can publish, in an automated way, the nightly. Is a good way to detect problems and bugs soon. In this case, is good to let the user change between the beta channel and the release channel in a clean way, and I suppose that this will need to use the same key for both (maybe I'm wrong, I never worked with the Play Store).
The protection against tampered comes with the keystore and the Play Store. The developer is responsible to have the private key in a secure place, and the user must install it only from the Play Store. If both things are accomplished the apk is secure and not tampered.
Another thing are developers or similar, we will build and install it using the keystore provided (or another one generated by us).

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by Asizon — DISMISSED** [review]

**Comment by mikeller** [comment]
@haslinghuis: Good point - one of them is #2574, which means I need to wait until this one has been merged before doing the rebase.

**Comment by mikeller** [comment]
@McGiverGim: I think for testing it is more helpful if we keep providing the .apk files that we are providing now - this allows the user to test with a specific version, or even with a build from a branch. If we publish the 'development' version in the Play store this will not be possible any more, unless the user first uninstalls the version from the store.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by Asizon — APPROVED** [review]

**Review by blckmn — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Pull request merged**
